app/src/main/java/com/gh4a/job/NotificationsJob.java

```diff
@@ -0,0 +1,129 @@
+package com.gh4a.job;
+
+import android.app.PendingIntent;
+import android.content.Intent;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.v4.app.NotificationCompat;
+import android.support.v4.app.NotificationManagerCompat;
+
+import com.evernote.android.job.Job;
+import com.evernote.android.job.JobManager;
+import com.evernote.android.job.JobRequest;
+import com.gh4a.Gh4Application;
+import com.gh4a.R;
+import com.gh4a.loader.NotificationListLoader;
+import com.gh4a.resolver.BrowseFilter;
+import com.gh4a.utils.ApiHelpers;
+import com.gh4a.utils.UiUtils;
+
+import org.eclipse.egit.github.core.Notification;
+import org.eclipse.egit.github.core.Repository;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class NotificationsJob extends Job {
+    private static final String GROUP_ID_GITHUB = "github_notifications";
+    private static final int ID_SUMMARY = 1;
+    public static final String TAG = "job_notifications";
+
+    public static void scheduleJob(int intervalMinutes) {
+        new JobRequest.Builder(TAG)
+                .setPeriodic(TimeUnit.MINUTES.toMillis(intervalMinutes),
+                        TimeUnit.MINUTES.toMillis(5))
+                .setRequiredNetworkType(JobRequest.NetworkType.CONNECTED)
+                .setRequirementsEnforced(true)
+                .setUpdateCurrent(true)
+                .build()
+                .schedule();
+    }
+
+    public static void cancelJob() {
+        JobManager.instance().cancelAllForTag(NotificationsJob.TAG);
+    }
+
+    @NonNull
+    @Override
+    protected Result onRunJob(Params params) {
+        try {
+            List<Notification> notifications =
+                    NotificationListLoader.loadNotifications(false, false);
+
+            if (notifications.isEmpty()) {
+                return Result.SUCCESS;
+            }
+
+            NotificationManagerCompat notificationManager =
+                    NotificationManagerCompat.from(getContext());
+            notificationManager.cancelAll();
+
+            int accentColor = UiUtils.resolveColor(getContext(), R.attr.colorAccent);
+
+            showSummaryNotification(notificationManager, notifications.size(), accentColor);
+            for (Notification notification : notifications) {
```

Comment:
I think it would be better to show one sub-notification per repo instead of per notification if the total number of notifications exceeds a certain threshold (e.g. 10).